httpcache.go

```diff
@@ -1,402 +1,407 @@
+		var e error
+		if e = combo.next.ServeHTTP(customWriter, combo.req); e != nil {
+			return e
+		}
 
-	errCh := make(chan error, 1)
-	go func() {
-		// This isn't necessary to shutdown Olric explicitly during Caddy's shutdown because Olric handles SIGTERM by itself
-		err = db.Start()
-		if err != nil {
-			c.logger.Error("olric.Start returned an error", zap.Error(err))
-			errCh <- err
+		combo.req.Response = customWriter.Response
+		if combo.req.Response, e = s.Retriever.GetTransport().(*rfc.VaryTransport).UpdateCacheEventually(combo.req); e != nil {
+			return e
 		}
-	}()
-	select {
-	case err = <-errCh:
-		return err
-	case <-started.Done():
-	}
-	dmap, err = db.NewDMap(dmapName)
-	return err
-}
 
-func (c *Cache) Validate() error {
-	if cfg == nil || c.Config == *cfg {
-		return nil
-	}
+		_, _ = customWriter.Send()
+		return e
+	})
 
-	return fmt.Errorf("the cache configuration is global and immutable, it must have the exact same value for all handlers")
+	return nil
 }
 
-func (c *Cache) writeResponse(w http.ResponseWriter, rdr io.Reader, fromCache bool) error {
-	// read the header and status first
-	var meta metadata
-	err := gob.NewDecoder(rdr).Decode(&meta)
-	if err != nil {
-		return err
+func (s *SouinCaddyPlugin) configurationPropertyMapper() error {
+	if s.Configuration != nil {
+		return nil
 	}
-
-	header := w.Header()
-	// set and write the cached headers
-	for k, v := range meta.Header {
-		header[k] = v
+	defaultCache := &DefaultCache{
+		Badger:      s.Badger,
+		Distributed: s.Olric.URL != "" || s.Olric.Path != "" || s.Olric.Configuration != nil,
+		Headers:     s.Headers,
+		Olric:       s.Olric,
+		TTL:         s.TTL,
 	}
-
-	if fromCache {
-		header.Set("Age", currentAge(meta.ResponseTime, meta.CorrectedInitialAge))
+	if s.Configuration == nil {
+		s.Configuration = &Configuration{
+			DefaultCache: defaultCache,
+			LogLevel:     s.LogLevel,
+		}
 	}
-
-	w.WriteHeader(meta.Status)
-
-	// write the cached response body
-	_, err = io.Copy(w, rdr)
-	return err
+	s.Configuration.DefaultCache = defaultCache
+	return nil
 }
 
-func (c *Cache) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddyhttp.Handler) error {
-	// TODO(dunglas): use functions added in https://github.com/pquerna/cachecontrol/pull/18 if merged
-	switch r.Method {
-	case http.MethodGet:
-	case http.MethodHead:
-	case http.MethodPost:
-	default:
-		// method not cacheable
-		w.Header().Add("Cache-Status", userAgent+"; fwd=request; detail=METHOD")
-		return next.ServeHTTP(w, r)
-	}
-
-	reqDir, err := cacheobject.ParseRequestCacheControl(r.Header.Get("Cache-Control"))
-	if err != nil || reqDir.NoCache || reqDir.NoStore {
-		// TODO: implement no-cache properly (add support for validation)
-		w.Header().Add("Cache-Status", userAgent+"; fwd=request; detail=DIRECTIVE")
-		return next.ServeHTTP(w, r)
-	}
+// Validate to validate configuration.
```

Comment:
If there is nothing to validate you can remove this method and the corresponding guard.